File: Python/Problems/Problems/problem3442.py
from .problem import Problem
import sys

class Problem3442(Problem):
    def test(self):
        s = "aaaaabbc"
        expected = 3

        result = self.maxDifference(s)

        return expected == result

    def maxDifference(self, s):
        charDic = {}
        for c in s:
            if (c in charDic):
                charDic[c] += 1
            else:
                charDic[c] = 1
        min = sys.maxsize
        max = 0
        for count in charDic.values():
            if count == 0:
                continue
            if count % 2 == 1 and count > max:
                max = count
            elif count % 2 == 0 and count < min:
                min = count

        return max - min

File: C++/Problems/Problems/problem_3442.cpp
#include "problem_3442.h"

#include <iostream>
#include <bits/ostream.tcc>

bool problem_3442::test() {
    auto n = "aaaaabbc";

    auto expected = 3;

    auto result = maxDifference(n);

    std::cout << result << std::endl;

    return expected == result;
}

int problem_3442::maxDifference(std::string s) {
    int charDic[26];
    for (int i = 0; i < 26; ++i)
    {
        charDic[i] = 0;
    }
    for (char c : s)
    {
        charDic[c - 'a']++;
    }

    auto min = INT_MAX;
    auto max = 0;
    for (auto count : charDic)
    {
        if (count == 0)
            continue;
        if (count % 2 == 1 && count > max)
            max = count;
        else if (count % 2 == 0 && count < min)
            min = count;
    }


    return max - min;
}
